azkaban-web-server/src/main/java/cloudflow/services/FlowServiceImpl.java

```diff
@@ -0,0 +1,200 @@
+package cloudflow.services;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+import azkaban.user.User;
+import cloudflow.daos.FlowDao;
+import cloudflow.daos.FlowVersionDao;
+import cloudflow.daos.ProjectDao;
+import cloudflow.error.CloudFlowException;
+import cloudflow.error.CloudFlowNotFoundException;
+import cloudflow.error.CloudFlowValidationException;
+import cloudflow.models.FlowResponse;
+import cloudflow.models.FlowVersion;
+import cloudflow.models.Project;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FlowServiceImpl implements FlowService {
+
+  private static final String PROJECT_ID_PARAM = "project_id";
+  private static final String PROJECT_VERSION_PARAM = "project_version";
+  private static final String PROJECT_NAME_PARAM = "project_name";
+  private static final String EXPAND_PARAM = "expand";
+  private static final String VERSIONS_KEY = "versions";
+  private static final Logger log = LoggerFactory.getLogger(ProjectServiceImpl.class);
+  private final FlowDao flowDao;
+  private final FlowVersionDao flowVersionDao;
+  private final ProjectDao projectDao;
+
+  @Inject
+  public FlowServiceImpl(FlowDao flowDao, FlowVersionDao flowVersionDao, ProjectDao projectDao) {
+    this.flowDao = flowDao;
+    this.projectDao = projectDao;
+    this.flowVersionDao = flowVersionDao;
+  }
+
+
+  @Override
+  public FlowResponse getFlow(String flowId, Map<String, String[]> queryParamMap) {
+    requireNonNull(flowId, "Flow id cannot be null");
```

Comment:
Recommend consistent casing for these messages across all new code. I have all these messages starting with lower-case chars.

Reply:
Will do this as a follow-up change to make it consistent for all new code.